app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context.rb

```diff
@@ -0,0 +1,190 @@
+require 'hawkularclient'
+
+class ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture
+  class CaptureContext
+    attr_reader :ts_values
+
+    def initialize(target, start_time, end_time, interval)
+      @target = target
+      @start_time = start_time || 15.minutes.ago.beginning_of_minute.utc
+      @end_time = end_time
+      @interval = interval
+      @tenant = target.try(:container_project).try(:name) || '_system'
+      @ext_management_system = @target.ext_management_system
+      @ts_values = Hash.new { |h, k| h[k] = {} }
+
+      if @target.respond_to?(:hardware)
+        hardware = @target.hardware
+      else
+        hardware = @target.container_node.hardware
+      end
+
+      @node_cores = hardware.try(:cpu_total_cores)
+      @node_memory = hardware.try(:memory_mb)
+
+      validate_target
+    end
+
+    def collect_metrics
+      case @target
+      when ContainerNode then collect_node_metrics
+      when Container then collect_container_metrics
+      when ContainerGroup then collect_group_metrics
+      else raise TargetValidationError, "Validation error: unknown target"
+      end
+    end
+
+    private
+
+    CPU_NANOSECONDS = 1e09
+
+    def target_name
+      "#{@target.class.name.demodulize}(#{@target.id})"
+    end
+
+    def validate_target
+      unless @ext_management_system
+        raise TargetValidationError, "Validation error: ems not defined"
+      end
+      unless @node_cores.to_i > 0
+        raise TargetValidationError, "Validation error: cores not defined"
+      end
+      unless @node_memory.to_i > 0
+        raise TargetValidationError, "Validation error: memory not defined"
+      end
```

Comment:
agree here ... it might push the line over the line length limit that rubocop likes, but I think it will read a little better